app/modules/asset_groups/models.py

```diff
@@ -607,234 +610,248 @@ def _get_pipeline_status_preparation(self):
+        ):
+            status['inProgress'] = True
+        # if it falls through, all False, thus "waiting"
+
         if progress.complete:
             status['end'] = progress.updated.isoformat() + 'Z'
         return status
 
-    # currently only gives most recent job
     def _get_pipeline_status_detection(self):
-        from app.utils import datetime_string_to_isoformat
+        from app.modules.progress.models import ProgressStatus
+
+        progress = self.progress_detection
+        # no progress object, we assume it has not yet started
+        if not progress:
+            return {
+                'skipped': False,
+                'inProgress': False,
+                'failed': False,
+                'complete': False,
+                'message': 'missing Progress',
+                'steps': 0,
+                'stepsComplete': 0,
+                'progress': 0,
+                'start': None,
+                'end': None,
+                'eta': None,
+                'ahead': None,
+                'status': None,
+                'description': None,
+            }
 
         status = {
+            # start with these false and set below
             'skipped': False,
-            'start': None,
-            'end': None,
             'inProgress': False,
+            'failed': False,
             'complete': False,
+            'message': progress.message,
+            'steps': len(progress.steps) if progress.steps else 0,
+            'stepsComplete': 0,  # TBD
+            'progress': progress.percentage / 100,
+            'start': progress.created.isoformat() + 'Z',
+            'end': None,
+            'eta': progress.current_eta,
+            'ahead': progress.ahead,
+            'status': progress.status,
+            'description': progress.description,
+            # these are provisional and should only be used for debugging, not ui/ux
+            # TODO i am not sure if this is the best count here, as there
+            #   is config['updatedAssets'] which may actually need to be considered
+            'numAssets': len(self.get_assets()),
+            'numAnnotations': len(self.get_all_annotations()),
+        }
+
+        if progress.skipped:
+            status['skipped'] = True
+        elif progress.status == ProgressStatus.failed:
+            status['failed'] = True
+        elif progress.complete:
+            status['complete'] = True
+        elif (
+            progress.status == ProgressStatus.created
+            or progress.status == ProgressStatus.healthy
+        ):
+            status['inProgress'] = True
+        # if it falls through, all False, thus "waiting"
+
+        if progress.complete:
+            status['end'] = progress.updated.isoformat() + 'Z'
+        return status
+
+    # this does not have a corresponding Progress, so we roll our own
+    def _get_pipeline_status_curation(self):
+        cur_start = self.get_curation_start_time()
```

Comment:
This doesn't seem to work with the way `get_curation_start_time is defined`. `cur_start` has a value when it shouldn't, for example, during the preparation stage.

Reply:
separate ticket for the first thing:  DEX-1236